Framework/Algorithms/src/Stitch1DMany.cpp

```diff
@@ -113,151 +176,248 @@ std::map<std::string, std::string> Stitch1DMany::validateInputs() {
+               m_scaleRHSWorkspace, m_useManualScaleFactor, m_manualScaleFactor,
+               lhsWS, outScaleFactor);
 
-    if (!isChild()) {
-      // Copy each input workspace's history into our output workspace's history
-      for (auto &inputWorkspace : m_inputWorkspaces)
-        lhsWS->history().addHistory(inputWorkspace->getHistory());
-    }
-    // We're a child algorithm, but we're recording history anyway
-    else if (isRecordingHistoryForChild() && m_parentHistory) {
-      m_parentHistory->addChildHistory(m_history);
-    }
+    m_scaleFactors.push_back(outScaleFactor);
+  }
+  m_outputWorkspace = lhsWS;
+
+  // Save output
+  this->setProperty("OutputWorkspace", m_outputWorkspace);
+  this->setProperty("OutScaleFactors", m_scaleFactors);
+}
 
-    m_outputWorkspace = lhsWS;
+/** Performs the Stitch1D algorithm at a specific workspace index.
+ * @param lhsWS :: The left-hand workspace to be stitched
+ * @param rhsWS :: The right-hand workspace to be stitched
+ * @param wsIndex :: The index of the rhs workspace being stitched
+ * @param startOverlaps :: Start overlaps for stitched workspaces
+ * @param endOverlaps :: End overlaps for stitched workspaces
+ * @param params :: Rebinning parameters
+ * @param scaleRhsWS :: Scaling either with respect to left or right workspaces
+ * @param useManualScaleFactor :: True to use a provided value for scale factor
+ * @param manualScaleFactor :: Provided value for scaling factor
+ * @param outWS :: Output stitched workspace
+ * @param outScaleFactor :: Actual value used for scale factor
+ */
+void Stitch1DMany::doStitch1D(
+    MatrixWorkspace_sptr lhsWS, MatrixWorkspace_sptr rhsWS, size_t wsIndex,
+    std::vector<double> startOverlaps, std::vector<double> endOverlaps,
+    std::vector<double> params, bool scaleRhsWS, bool useManualScaleFactor,
+    double manualScaleFactor, MatrixWorkspace_sptr &outWS,
+    double &outScaleFactor) {
+
+  IAlgorithm_sptr alg = createChildAlgorithm("Stitch1D");
+  alg->initialize();
+  alg->setProperty("LHSWorkspace", lhsWS);
+  alg->setProperty("RHSWorkspace", rhsWS);
+  if (startOverlaps.size() > wsIndex - 1) {
+    alg->setProperty("StartOverlap", startOverlaps[wsIndex - 1]);
+    alg->setProperty("EndOverlap", endOverlaps[wsIndex - 1]);
+  }
+  alg->setProperty("Params", params);
+  alg->setProperty("ScaleRHSWorkspace", scaleRhsWS);
+  alg->setProperty("UseManualScaleFactor", useManualScaleFactor);
+  if (useManualScaleFactor)
+    alg->setProperty("ManualScaleFactor", manualScaleFactor);
+  alg->execute();
+
+  if (!isChild()) {
+    // Copy each input workspace's history into our output workspace's history
+    for (auto &inputWS : m_inputWSMatrix[wsIndex])
+      lhsWS->history().addHistory(inputWS->getHistory());
+  }
+  // We're a child algorithm, but we're recording history anyway
+  else if (isRecordingHistoryForChild() && m_parentHistory) {
+    m_parentHistory->addChildHistory(m_history);
```

Comment:
If I run `Stitch1DMany` on matrix workspaces (i.e. no groups):

```
x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
y = [1, 1, 1, 1, 1, 1, 1, 1, 1]
ws_1_1 = CreateWorkspace(x, y)

x = [9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
y = [2, 2, 2, 2, 2, 2, 2, 2, 2]
ws_2_1 = CreateWorkspace(x, y)

x = [17, 18, 19, 20, 21, 22, 23, 24, 25, 26]
y = [3, 3, 3, 3, 3, 3, 3, 3, 3]
ws_3_1 = CreateWorkspace(x, y)

stitched, scale_factors = Stitch1DMany(InputWorkspaces='ws_1_1, ws_2_1, ws_3_1', Params='1')
```

On master the history contains also `CreateWorkspace v.1`, but on this branch only `Stitch1DMany v.1` appears.